api/utils/singlepoint_helper.py

```diff
@@ -40,41 +41,59 @@ def convert_ndarray_to_list(
 
 def singlepoint(
     struct: Path,
-    arch: Architectures | None = "mace_mp",
-    properties: list[Properties] | None = None,
-    range_selector: str | None = ":",
-) -> dict[str, Any]:
+    arch: Architectures = "mace_mp",
+    properties: list[Properties] = None,
+    range_selector: str = ":",
+    write_results: bool = True,
+    results_path: Path = DATA_DIR / "results/",
+    format: str = "extxyz",
+) -> SinglePointResults:
     """
     Perform single point calculations and return results.
 
     Parameters
     ----------
-    struct : str
-        Filename of structure to simulate.
+    struct : Path
+        Path of structure to simulate.
     arch : Architectures
         MLIP architecture to use for single point calculations. Default is "mace_mp".
     properties : List[Properties]
         Physical properties to calculate. Default is ("energy", "forces", "stress").
     range_selector : str
         Range of indices to include from the structure. Default is all.
+    write_results : bool | None, default is True
+        Tells function if to save the results of the calculation or not.
```

Comment:
```suggestion
    write_results : bool
        Tells function if to save the results of the calculation or not. Default is True.
```
(Check if it's too long)

Reply:
Fixed